android/source/VideoLocker/src/org/edx/mobile/http/HttpManager.java

```diff
@@ -133,8 +135,25 @@ public String post(String url, Bundle params, Bundle headers)
                 } catch (JSONException e) {
                     e.printStackTrace();
                 }
+
+                // end connection and return
+                client.getConnectionManager().shutdown();
                 return json.toString();
             }
+        } else if (statusCode == 401) {
+            // for google/FB login, this means google/FB account is not associated with edX
+            LogUtil.log(TAG, "HTTP 401");
+
+            JSONObject json = new JSONObject();
+            try {
+                json.put("error", "401");
+            } catch (JSONException e) {
+                e.printStackTrace();
```

Comment:
There's basically no way this can fail, and you're just checking it because it's required, right? Maybe worth leaving a comment.